app/src/main/cpp/skyline/kernel/memory.cpp

```diff
@@ -61,7 +61,9 @@ namespace skyline::kernel {
         if (!base.address)
             throw exception("Cannot find a suitable carveout for the guest address space");
 
-        mmap(reinterpret_cast<void *>(base.address), base.size, PROT_NONE, MAP_ANONYMOUS | MAP_PRIVATE, -1, 0);
+        auto ret{mmap(reinterpret_cast<void *>(base.address), base.size, PROT_NONE, MAP_FIXED | MAP_ANONYMOUS | MAP_PRIVATE, -1, 0)};
+        if (ret != reinterpret_cast<void *>(base.address))
```

Comment:
```suggestion
        auto result{mmap(reinterpret_cast<void *>(base.address), base.size, PROT_NONE, MAP_FIXED | MAP_ANONYMOUS | MAP_PRIVATE, -1, 0)};
        if (result == MAP_FAILED) [[unlikely]]
```
I should mention I will be updating the error if clauses for other system calls with `[[unlikely]]` in another PR

Reply:
Is there really any point? Like it really isn't going to have any effect on performance aside from making code more cluttered

Reply:
It isn't on a small scale but optimizing branches will have an advantage on a larger scale, it is why projects like the Linux kernel or even Atmosphere do them as well. This also includes things such as arranging branches in the order of probabilities, putting an error branch here like this ends up making it the most probable and puts the related assembly right after it while it should really be in the function epilogue.

Reply:
Smh fine